### PR TITLE
Getting model source value

### PR DESCRIPTION
**Pull request by Malikbadmus**
Thank you for taking your time to contribute to Ersilia, just a few checks before we proceed
- [x] Have you followed the guidelines in our [Contribution Guide](https://github.com/ersilia-os/ersilia/blob/master/CONTRIBUTING.md)
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

**Description**

Keeping track of where a model was fetched from, whether a repository or storage location seems crucial, a model source descriptor was created to store these values.

**Changes to be made**

- Created a model_source file.
- Added a  `_get_model_source` method to the information class.
- Included model source in the data response payload in the get method.
- Modified the code to also include local path.

**Status**

- The get method returns a data response payload that includes the model source, along with other attributes such as pack mode, service class, API schema, size, metadata, and card."
- Information.json file now includes model source in its data object.
- Model source is not currently reflected in information.json for models fetched from DockerHub.

**To do**

- Create Datastore for Ersilia catalog.
- Reflect correct memory, service class, and model source information in the information.json file, particularly in the case of a model getting fetched from DockerHub.

Related to #1196

### CODE REVIEW

**Comment by Malikbadmus** [comment]
@DhanshreeA , this is how the information.json looks like now.

[information.json](https://github.com/user-attachments/files/16275091/information.json)

File: ersilia/hub/fetch/fetch.py [hunk 1]
```diff
@@ -63,6 +64,22 @@ def __init__(
...
+            self.force_from_s3: "Amazon S3",
+            self.force_from_dockerhub: "DockerHub",
+            self.force_from_hosted: "Hosted services",
+            self.force_with_bentoml: "Bentoml",
+            self.force_with_fastapi: "Fastapi",
+            self.hosted_url is not None: "Hosted URL",
+            self.local_dir is not None: "Local path"
+        }
+        
+        self.model_source = next((source for condition, source in sources.items() if condition), "DockerHub")
```
Comment (DhanshreeA) [inline ersilia/hub/fetch/fetch.py:81]: This is a really neat implementation, I like it!

**Review by DhanshreeA — APPROVED** [review]

**Comment by DhanshreeA** [comment]
LGMT @Malikbadmus , thanks!

**Comment by Malikbadmus** [comment]
> LGMT @Malikbadmus , thanks! Have you tested this with the catalog command? 

Yes I have.

**Pull request merged**